tests/integ/scala/test_update_delete_merge_suite.py

```diff
@@ -367,39 +368,40 @@ def test_merge_with_insert_clause_only(session, local_testing_mode):
         ],
         schema=StructType([StructField("id", StringType())]),
     )
-    assert target.merge(
-        source_df,
-        target["id"] == source_df["id"],
-        [
-            when_not_matched().insert(
-                {
-                    "id": source_df["id"],
-                    "col_datetime": current_timestamp(),
-                    "col_date": current_date(),
-                    "col_time": current_time(),
-                }
-            )
-        ],
-    ) == MergeResult(2, 0, 0)
-    res = target.collect()
-    assert (
-        len(res) == 3
-        and res[0][0] == "id1"
-        and res[0][1] == fixed_date
-        and res[0][2] == fixed_date.date()
-        and res[0][3] == fixed_date.time()
-    )
-    assert res[1][0] == "id2" and res[2][0] == "id3"
-
-    if local_testing_mode:
-        assert res[1][1].date() == res[2][1].date() == now_datetime.date()
-        assert res[1][2] == res[2][2] == now_datetime.date()
-        assert res[1][3].minute == res[2][3].minute == now_datetime.time().minute
-        assert res[1][3].hour == res[2][3].hour == now_datetime.time().hour
-    else:
-        # in live connection time can differ because server timezone and test machine timezone differ
-        # so that we do not test exact time
-        assert len(res[1]) == len(res[2]) == 4
+    with patch("snowflake.snowpark.mock._functions.datetime") as dt:
```

Comment:
it is kind of wired that we are using mock for integration test? what is this part of code trying to test? does it has to use datetime.now()?

Reply:
or maybe you can move this to unit test if possible?

Reply:
@sfc-gh-yzou  you have a valid point, this test is overloaded for multiple purpose -- testing merge + current timeout implementation in local testing and could be decoupled.

I will refactor it a bit

Reply:
updated